EDMarketConnector.py

```diff
@@ -18,74 +18,69 @@
-            ShellExecute.argtypes = [HWND, LPCWSTR, LPCWSTR, LPCWSTR, LPCWSTR, INT]
-
-            def window_title(h):
-                if h:
-                    text_length = GetWindowTextLength(h) + 1
-                    buf = ctypes.create_unicode_buffer(text_length)
-                    if GetWindowText(h, buf, text_length):
-                        return buf.value
-
-                return None
-
-            @ctypes.WINFUNCTYPE(BOOL, HWND, LPARAM)
-            def enumwindowsproc(window_handle, l_param):
-                # class name limited to 256 - https://msdn.microsoft.com/en-us/library/windows/desktop/ms633576
-                cls = ctypes.create_unicode_buffer(257)
-                if GetClassName(window_handle, cls, 257) \
-                        and cls.value == 'TkTopLevel' \
-                        and window_title(window_handle) == applongname \
-                        and GetProcessHandleFromHwnd(window_handle):
-                    # If GetProcessHandleFromHwnd succeeds then the app is already running as this user
-                    if len(sys.argv) > 1 and sys.argv[1].startswith(protocolhandler.redirect):
-                        CoInitializeEx(0, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE)
-                        # Wait for it to be responsive to avoid ShellExecute recursing
-                        ShowWindow(window_handle, SW_RESTORE)
-                        ShellExecute(0, None, sys.argv[1], None, None, SW_RESTORE)
-
-                    else:
-                        ShowWindowAsync(window_handle, SW_RESTORE)
-                        SetForegroundWindow(window_handle)
-
-                    return False
-
-                return True
-
-            return EnumWindows(enumwindowsproc, 0)
+            print('no_other_instance_running(): win32, using msvcrt')
+            # win32 doesn't have fcntl, so we have to use msvcrt
+            import msvcrt
+
+            print(f'no_other_instance_running(): journal_dir_lockfile = {journal_dir_lockfile!r}')
+
+            try:
+                msvcrt.locking(journal_dir_lockfile.fileno(), msvcrt.LK_NBLCK, 4096)
+
+            except PermissionError as e:
+                print(f"PermissionError: Couldn't lock journal directory \"{journal_dir}\", assuming another process running\n{e}")
+                return False
 
+            except OSError as e:
+                print(f"OSError: Couldn't lock journal directory \"{journal_dir}\", assuming another process running\n{e}")
+                return False
+
+            except Exception as e:
+                print(f"other Exception: Couldn't lock journal directory \"{journal_dir}\", assuming another process running\n{e}")
+                return False
+
+            journal_dir_lockfile.write(f"Path: {journal_dir}\nPID: {os_getpid()}\n")
+
+        else:
+            print('no_other_instance_running(): NOT win32, using fcntl')
+            try:
+                import fcntl
+
+            except Exception:
```

Comment:
```suggestion
            except ImportError:
```
Rather only catch the "It doesnt exist" case